paasta_tools/cli/cmds/status.py

```diff
@@ -1253,28 +1253,34 @@ def apply_args_filters(
     ] = defaultdict(lambda: defaultdict(dict))
 
     if args.service:
+        if '.' in args.service:
+            service_name = args.service[:args.service.find('.')]
+            args.instances = args.service[args.service.find('.')+1:]
+            args.service = service_name
```

Comment:
semi-subtle and rather nitpicky point: I kind of prefer writing to a new variable like `service_name` rather than overwriting something on `args`.  But if we're going to do that, we should use `instance_name` or something similar instead of overwriting `args.instance`.    

I feel like it's not great to mutate your arguments if you can avoid it, and also if you overwrite `args.instance` or `args.service` then it makes the code a little harder to follow -- below we do more things with `args.instance`, and when I see `args.instance`, my impression is that those are the args as they were passed in.

Reply:
https://github.com/Yelp/paasta/blob/d7689065a90cea73c81b39af5e5e287d1c938327/paasta_tools/cli/cmds/status.py#L1289

@stug Good point. The reason why I altered `args` directly was because this function calls `get_filters(args)`. 

https://github.com/Yelp/paasta/blob/d7689065a90cea73c81b39af5e5e287d1c938327/paasta_tools/cli/cmds/status.py#L1186-L1191 

Would you suggest I alter the signature of this function entirely to support things like `service_name`, `instance_name`, and etc?

Reply:
If you have time that would be nice, but this makes things a bit more complicated so feel free to punt.